thirdparty/CMakeLists.txt

```diff
@@ -108,6 +108,7 @@ ExternalProject_Add(gperftools
 set(HDFS_CLIENT_DIR "hadoop-hdfs-project/hadoop-hdfs-native-client")
 ExternalProject_Add(hadoop
         URL https://github.com/apache/hadoop/archive/refs/tags/rel/release-2.8.4.tar.gz
+        https://pegasus-resource-package.oss-cn-beijing.aliyuncs.com/hadoop-release-2.8.4.tar.gz
```

Comment:
> List of paths and/or URL(s) of the external project's source. When more than one URL is given, they are tried in turn until one succeeds.

So, better to keep URLs in order, aliyun OSS URL at first, then original URL, most developers are in China mainland after all.

Reply:
I have talked with @neverchanje , you can add another aliyun oss url and define a new OSS_URL_PREFIX.